### PR TITLE
[UI] Improve drag-and-drop support in APL editor

### PR DESCRIPTION
**Pull request by hillerstorm**
Doing a ton of APL editing highlighted a few missing features that would really speed things up, so I added them!

* Drag and drop APL values inside an `Any of` / `All of`
  * Not restricted to the same list, can be moved in/out any number of levels **and even between different APL actions**
* Being able to copy an APL value inside an `Any of` / `All of`
* Clearer indication of what block will be deleted inside nested `Any of` / `All of`
* Having the whole block show up as the drag-placeholder instead of just the move-button

Any feedback is appreciated!

https://github.com/user-attachments/assets/20f6edd3-4b19-4cfd-bee7-7a4b00f79ddf

### CODE REVIEW

File: ui/core/components/pickers/list_picker.tsx [hunk 1]
```diff
@@ -246,10 +282,14 @@ export class ListPicker<ModObject, ItemType> extends Input<ModObject, Array<Item
 			itemContainer.addEventListener(
 				'dragenter',
 				event => {
-					if (!curDragData || curDragData.listPicker != this) {
+					if (
+						!curDragData ||
+						curDragData.listPicker.config.itemLabel !== this.config.itemLabel ||
```
Comment (1337LutZ) [inline ui/core/components/pickers/list_picker.tsx:287]: Can this specific if check become a helper function? I see it repeated a lot (not that the old one wasn't)

File: ui/core/components/pickers/list_picker.tsx [hunk 2]
```diff
@@ -274,30 +318,73 @@ export class ListPicker<ModObject, ItemType> extends Input<ModObject, Array<Item
...
+						(this.config.itemLabel === 'Action' && curDragData.listPicker !== this)
+					) {
+						return;
+					}
+					event.stopPropagation();
+					itemContainer.removeAttribute('draggable');
+					curDragData.item.elem.removeAttribute('draggable');
+					curDragData.item.elem.classList.remove('dragfrom');
+					[...document.querySelectorAll('.dragto,.hover')].forEach(elem => {
```
Comment (1337LutZ) [inline ui/core/components/pickers/list_picker.tsx:349]: Getting these specific values can probably also become a getter/util

File: ui/core/components/pickers/list_picker.tsx (hunk 1 above)
Comment (hillerstorm) [reply ui/core/components/pickers/list_picker.tsx:287]: Yea I could do that 👍

File: ui/core/components/pickers/list_picker.tsx (hunk 2 above)
Comment (hillerstorm) [reply ui/core/components/pickers/list_picker.tsx:349]: Do you mean the classes? Like a helper to fetch any number of elements matching certain classes and then remove the classes from the matching elements?
A very niche use case :D

Comment (hillerstorm) [reply ui/core/components/pickers/list_picker.tsx]: Cleaned up some repetitive code, this one being one of them

File: ui/core/components/pickers/list_picker.tsx [hunk 3]
```diff
@@ -277,16 +281,18 @@ export class ListPicker<ModObject, ItemType> extends Input<ModObject, Array<Item
 				},
 				{ signal: this.signal },
 			);
+			
+			const invalidDropTarget = () => {
```
Comment (1337LutZ) [inline ui/core/components/pickers/list_picker.tsx:285]: Yes, this is what I meant :D

**Review by 1337LutZ — APPROVED** [review]
I haven't tested the feature myself, but code wise it looks cleaner :D thanks for making some of the small util fns

**Pull request merged**